shotover-proxy/tests/cassandra_int_tests/prepared_statements.rs

```diff
@@ -102,6 +102,12 @@ async fn use_statement(session: &CassandraConnection) {
     .await;
 }
 
+async fn hardcoded_prepared(session: &CassandraConnection) {
```

Comment:
This is testing a bug fix that is separate to the main goal of using replication_factor in our routing, can this test and the bug fix be moved into a separate PR?

Reply:
As part of implementing the keyspace info I added error handling which fixes that bug so I included that in this PR.